src/database/firebase.tsx

```diff
@@ -18,6 +19,149 @@ const config = {
   appId: APP_ID,
 };
 
-firebase.initializeApp(config);
+!firebase.apps.length ? firebase.initializeApp(config) : firebase.app();
+
+const database = firebase.firestore();
+const treeCollection = database.collection('trees');
+const commentCollection = database.collection('comments');
+const additionalCollection = database.collection('additional');
+
+/*
+ * checkID validates that this ID exists in the `trees` table.
+ */
+export const checkID = async function (id: string): Promise<boolean> {
+  await treeCollection
+    .doc(id)
+    .get()
+    .then(doc => {
+      return doc.exists;
+    })
+    .catch(error => {
+      console.log('Error getting document:', error);
+    });
+  return false;
+};
+
+/*
+ * getTree queries the `trees` table and returns a Tree if the ID is found and an empty entry otherwise.
+ */
+export const getTree = async function (id: string): Promise<Tree> {
+  await treeCollection
+    .doc(id)
+    .get()
+    .then(doc => {
+      return doc.data() as Tree;
+    })
+    .catch(error => {
+      console.log('Error getting document:', error);
+    });
+  return { id: '' } as Tree;
+};
+
+/*
+ * getAllTrees returns a Dictionary containing all entries in the `trees` table.
+ */
+export const getAllTrees = async function (): Promise<Dictionary> {
```

Comment:
Why return a Dictionary? Seems more intuitive to have the default return an array of trees but maybe rename this to like `getTreesDictionary` or smth like that?

Reply:
changed to array